jdk17-unit-tests/build.gradle

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+plugins {
+  id 'java-library'
+  // For code coverage:
+  id 'jacoco'
+  id 'com.github.nbaztec.coveralls-jacoco'
+}
+
+// Use JDK 17 for this module, via a toolchain
+java.toolchain.languageVersion.set JavaLanguageVersion.of(17)
+
+dependencies {
+    testImplementation project(":nullaway")
+    testImplementation deps.test.junit4
+    testImplementation(deps.build.errorProneTestHelpers) {
+        exclude group: "junit", module: "junit"
+    }
+    testImplementation deps.build.jsr305Annotations
+}
+
+test {
+    maxHeapSize = "1024m"
+    // to expose necessary JDK types on JDK 16+; see https://errorprone.info/docs/installation#java-9-and-newer
+    jvmArgs += [
+            "--add-exports=jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.main=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.model=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.processing=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED",
+            "--add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED",
+            "--add-opens=jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED",
+            "--add-opens=jdk.compiler/com.sun.tools.javac.comp=ALL-UNNAMED",
+    ]
+}
+
+jacoco {
+  toolVersion = "0.8.7"
+}
+
+jacocoTestReport {
+    reports {
+        xml.enabled = true // coveralls plugin depends on xml format report
+        html.enabled = true
+    }
+}
+
+coverallsJacoco {
+    reportPath = "jdk17-unit-tests/build/reports/jacoco/test/jacocoTestReport.xml"
```

Comment:
One issue here is that our overall coverage info will be split between the unit tests in the `:nullaway` module and the tests here.  I wonder if there is some way to merge this information before sending to Coveralls @lazaroclapp?  Already with the previous version of this PR I was getting bad warnings about reduced coverage, since the coverage info from the new tests here wasn't getting merged in.